src/client/src/OpenFin/apps/Launcher/NlpSuggestions/executions/RfqExecution/state.ts

```diff
@@ -0,0 +1,81 @@
+import { bind } from "@react-rxjs/core"
+import { createSignal } from "@react-rxjs/utils"
+import { concat, exhaustMap, map, switchMap, take, withLatestFrom } from "rxjs"
+
+import { CREDIT_RFQ_EXPIRY_SECONDS } from "@/constants"
+import {
+  createCreditRfq$,
+  creditDealers$,
+  creditInstruments$,
+} from "@/services/credit"
+
+import { nlpIntent$, NlpIntentType } from "../../../services/nlpService"
+import { NlpExecutionDataReady, NlpExecutionStatus } from "../types"
+
+const [next$_, onNext] = createSignal()
+export { onNext }
+
+const next$ = next$_.pipe(take(1))
+
+const rfqExecutionState$ = nlpIntent$.pipe(
+  withLatestFrom(creditInstruments$, creditDealers$),
+  switchMap(([intent, instruments, dealers]) => {
+    if (
+      !intent ||
+      intent === "loading" ||
+      intent.type !== NlpIntentType.CreditRfq ||
+      !intent.payload.symbol ||
+      !intent.payload.direction ||
+      !intent.payload.notional ||
+      !dealers.length
+    ) {
+      return [{ type: NlpExecutionStatus.MissingData as const, payload: {} }]
+    }
+
+    const requestData =
+      intent.payload as NlpExecutionDataReady["payload"]["requestData"]
+    const { symbol, direction, notional } = requestData
+
+    const instrument = instruments.find(
+      (instrument) => instrument.ticker === symbol,
+    )
+
+    if (!instrument) {
+      return [{ type: NlpExecutionStatus.MissingData as const, payload: {} }]
+    }
+
+    return concat(
+      [
+        {
+          type: NlpExecutionStatus.DataReady as const,
+          payload: { requestData },
```

Comment:
We never use any of these payloads except for the Done one, probably worth at least removing the requestData from them

Reply:
I do use the payload for waiting to execute so ill just remove the request data for the dataReady